src/controllers/products.js

```diff
@@ -22,6 +22,22 @@ exports.createProduct = async (req, res, next) => {
   return res.status(201).json({ success: true, data: product });
 };
 
+// check if requestedBuyers doesn't have buyerId already. product.requestedBuyers.find(i => i.toHexString() === buyerId.toHexString())
+exports.requestProduct = async (req, res, next) => {
+  const productId = req.params.id;
+  const buyerId = req.user._id;
+  const product = await Product.findOne({ _id: productId });
+  // check if the product is not null
+  if (product) {
+    product.requestedBuyers.push(buyerId);
```

Comment:
Before pushing it again, you might want to check, if the user is already among one of the requestedBuyers. I mean try executing this request multiple times, and check the `requestedBuyers` field.